pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_reconcile.go

```diff
@@ -159,18 +161,33 @@ func (c *controller) reconcilePermissionClaims(ctx context.Context, apiBinding *
 		}
 	}
 
-	klog.V(2).Infof("Adding permission claims %v for APIBinding %s", addedPermissionClaims, fmt.Sprintf("%s|%s", lc, apiBinding.Name))
+	if len(addedPermissionClaims) != 0 {
+		klog.V(2).Infof("Adding permission claims %v for APIBinding %s", addedPermissionClaims, fmt.Sprintf("%s|%s", lc, apiBinding.Name))
+	}
 
 	var allPatchErrors []error
 	for gvr, claims := range addedPermissionClaims {
 		for _, claim := range claims {
 			//TODO: use indexer instead of lister.
-			objs, err := claim.lister.Lister().List(labels.Everything())
+			if !claim.informer.Informer().HasSynced() {
+				return fmt.Errorf("unable sync cache")
+			}
+			// objs, err := claim.informer.Informer().GetIndexer().ByIndex(indexers.ByLogicalCluster, lc.String())
+			// if err != nil {
+			// 	return err
+			// }
```

Comment:
left over?

Actually, am not sure how an indexer would actually do this. We cannot find those that are NOT assigned to an indexer key.

Reply:
removed